File: Npc/11000101.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000101: Ray
/// </summary>
public class _11000101 : NpcScript {
    protected override int First() {
        return 20;
    }

    // Select 0:
    // $script:0831180407000396$
    // - Yo, man! Wassup!
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (20, 0):
                // $script:0831180407000398$
                // - Hey yo, do you know what hip-hop represents?
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (20, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000101.py
""" 11000101: Ray """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 20

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407000396$
        # - Yo, man! Wassup!
        return None # TODO

    def __20(self, pick: int) -> int:
        # $script:0831180407000398$
        # - Hey yo, do you know what hip-hop represents?
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (20, 0):
            return Option.CLOSE
        return Option.NONE
